### PR TITLE
achievement types update properties

### PR DESCRIPTION
**Pull request by Northburns**
Reference <https://wiki.guildwars2.com/wiki/API:2/achievements>

* `AchievementBit`: API spec says type is required, but at least Dragon Ice Infuser returns array of empty bit objects
* `Achievement`, the _Walking on Fire_ doesn't have rewards attached to it, so it now has a default value of empty list.

### CODE REVIEW

File: src/test/kotlin/io/github/kryszak/gwatlin/api/achievement/AchievementsClientTest.kt [hunk 1]
```diff
@@ -6,6 +6,8 @@ import io.kotest.assertions.assertSoftly
 import io.kotest.assertions.throwables.shouldThrow
 import io.kotest.matchers.collections.shouldContainExactly
 import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.collections.shouldMatchEach
```
Comment (Kryszak) [inline src/test/kotlin/io/github/kryszak/gwatlin/api/achievement/AchievementsClientTest.kt:9]: Seems like unused import

File: src/main/kotlin/io/github/kryszak/gwatlin/api/achievement/model/Achievement.kt [hunk 2]
```diff
@@ -16,6 +16,6 @@ data class Achievement(
         val type: String,
         val flags: List<String>,
```
Comment (Kryszak) [inline src/main/kotlin/io/github/kryszak/gwatlin/api/achievement/model/Achievement.kt:17]: Let's initialize `flags` and `tiers` with empty list too. Might save us revisiting this model later on :smile:

**Review by Kryszak — CHANGES_REQUESTED** [review]

File: src/test/kotlin/io/github/kryszak/gwatlin/api/achievement/AchievementsClientTest.kt (hunk 1 above)
Comment (Northburns) [reply src/test/kotlin/io/github/kryszak/gwatlin/api/achievement/AchievementsClientTest.kt:9]: Good catch. Seems I've grown reliant on linters 🙈

File: src/main/kotlin/io/github/kryszak/gwatlin/api/achievement/model/Achievement.kt (hunk 2 above)
Comment (Northburns) [reply src/main/kotlin/io/github/kryszak/gwatlin/api/achievement/model/Achievement.kt]: Good call, done.

File: src/test/kotlin/io/github/kryszak/gwatlin/api/achievement/AchievementsClientTest.kt (hunk 1 above)
Comment (Kryszak) [reply src/test/kotlin/io/github/kryszak/gwatlin/api/achievement/AchievementsClientTest.kt]: Like we all, like we all :smile:

**Review by Kryszak — APPROVED** [review]

**Pull request merged**